### PR TITLE
Setup Husky, eslint and pre-commit hook

### PR DESCRIPTION
**Pull request by SBagaria2710**
Added Husky and Lint-Staged to lint staged files before committing.

Fixes: https://github.com/Real-Dev-Squad/website-learn/issues/73

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (samarpan1738) [inline .vscode/settings.json:1]: Should these settings be shared across dev working on this repo? If no then we can remove this file

File: package.json [hunk 2]
```diff
@@ -48,6 +52,19 @@
...
+      "pre-commit": "npx lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/*.{js,vue}": [
+      "eslint --fix",
+      "vue-cli-service lint",
+      "git add"
+    ],
+    "./**": ["git add"]
```
Comment (samarpan1738) [inline package.json:66]: What does this line do?

File: package.json [hunk 3]
```diff
@@ -27,12 +28,15 @@
     "eslint": "^6.7.2",
     "eslint-plugin-import": "^2.20.2",
     "eslint-plugin-vue": "^7.0.0",
+    "husky": "^8.0.1",
```
Comment (samarpan1738) [inline package.json:31]: Shouldn't husky and lint-staged be a dev-dependency?

**Review by samarpan1738 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (SBagaria2710) [reply package.json:31]: They are in devDependency only 🤔

File: package.json (hunk 2 above)
Comment (SBagaria2710) [reply package.json]: I though the root level files weren't getting committed for cases when linter found something to fix.
Just tested and was able to do it now, so removed this line altogether.

File: .vscode/settings.json (hunk 1 above)
Comment (SBagaria2710) [reply .vscode/settings.json:1]: Forgot to add this in the `.gitignore` 🤦🏼‍♂️

**Review by Gautam-Arora24 — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (samarpan1738) [reply package.json:31]: My bad

File: .vscode/settings.json (hunk 1 above)
Comment (samarpan1738) [reply .vscode/settings.json:1]: Even though .vscode is added in .gitignore, still .vscode folder is not removed

**Review by samarpan1738 — APPROVED** [review]
LGTM 👍

**Pull request merged**